lib/util/tests/ert_util_normal_path.c

```diff
@@ -0,0 +1,71 @@
+/*
+   Copyright (C) 2017  Statoil ASA, Norway.
+
+   The file 'ert_util_normal_path.c' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#include <stdlib.h>
+#include <stdbool.h>
+
+#include <ert/util/test_util.h>
+#include <ert/util/test_work_area.h>
+#include <ert/util/util.h>
+
+
+void test_path(const char * input_path, const char * expected_path) {
+  char * normal_path = util_alloc_normal_path( input_path );
+  test_assert_string_equal( normal_path , expected_path );
+  free( normal_path );
+}
+
+
+void test_relative() {
+  test_work_area_type * work_area = test_work_area_alloc("Work");
+  util_make_path("level0/level1/level2");
+
+
+  test_path( "level0/level1/../", "level0");
+  test_path( "level0", "level0");
+  test_path( "level0/././level1/../", "level0");
+  test_path( "level0/level1/../level1/level2/../", "level0/level1");
+  test_path( "level0/level1/../level1/level2/..", "level0/level1");
+  test_path( "level0/level1/../level1/level2/../file.txt", "level0/level1/file.txt");
```

Comment:
Please add a test that does not move linearly in the file system. Something like:
`test_path( "level0/level1/../level1a", "level0/level1a" );`
I expect it to pass, but I would like it to there.

Reply:
New test added and force-pushed

Reply:
Btw, are these tests ran at Windows as well? I would expect them to fail due to the forward slash and `util_path_sep` using `UTIL_PATH_SEP_STRING` to split files.

Reply:
Ehh - no; as commented here: https://github.com/Statoil/libecl/pull/232#discussion_r149655688